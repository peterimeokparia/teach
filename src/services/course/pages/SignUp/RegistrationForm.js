import React, { 
useState } from 'react';

import { 
Validations } from  '../../../../helpers/validations';

import { 
toast } from 'react-toastify';

import './LoginPage.css';



const RegistrationForm = ({ 
error, 
loading,
handleCreateUser, 
users }) => {

  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [firstName, setFirstName] = useState('');
  const [userRole, setUserRole] = useState('');
  const [ displayOtherFormFields, setToDisplayOtherFormFields ] = useState(false);


  const roles = {
       Tutor: "Tutor",
       Student: "Student" 
  }
   
  
  let currentUser = null;


   const handleSubmit = (e) => {
      e.preventDefault();
   }


   const getOtherFormFields = () => {

      if ( email ) {

          if ( users?.find(user => user?.email === email ) === undefined ) {

               setToDisplayOtherFormFields( true );
  
             
          } else {

              toast.error("Account already exists in our system.");
              return;


          }
     
      }
     
   }

  
  return   (    
            <div className="LoginPage"> 

              <form onSubmit={ e => handleSubmit(e)}>
             {
                  <>
                          <label>  
                        
                            Email

                            <input
                                name="email"
                                type="email"
                                value={email}
                                onChange={ e => setEmail( e.target.value ) }
                                placeholder="email"
                            >
                            </input>
                          </label>

                          {

                              ( displayOtherFormFields ) ? ""

                                                         : <button
                                                              type="submit"
                                                              disabled={loading}
                                                              onClick={getOtherFormFields}
                                                          >
                                                                Next

                                                          </button>     


                          }
                                            
                                             
              </>


             }
                 
              {
                ( displayOtherFormFields ) &&  <>
                   <label>

                    Password   

                    <input
                        name="password"
                        type="password"
                        value={password}
                        onChange={ e => setPassword( e.target.value ) }
                        placeholder="password"
                    >
                    </input>
                  </label>


                  <label>

                    First Name   

                    <input
                        name="firstName"
                        type="text"
                        value={firstName}
                        onChange={ e => setFirstName( e.target.value ) }
                        placeholder="first name"
                    >
                    </input>
                  </label>

    
                  <span className="LoginPageRadioButton">

                  <span className="left">

                      <label> 
                        
                        Student

                      <input
                            name="firstName"
                            type="radio"
                            value={roles.Student}
                            onChange={ e => setUserRole( e.target.value ) }
                            placeholder="first name"
                        >
                        </input>
                      </label>
                    
                  </span>  

                  
                 <span className="right">

                 <label>

                    Tutor

                    <input
                        name="firstName"
                        type="radio"
                        value={roles.Tutor}
                        onChange={ e => setUserRole( e.target.value ) }
                        placeholder="first name"
                    >
                    </input> 

                  </label>  

                 </span>
                 
                   
                  </span>

                  <div>
                                  
                    </div>
  
                      { error  && (<div className="error"> { error.message }</div>)}
  
                        <div></div>
                        
                          <button
                                type="submit"
                                disabled={loading}
                                onClick={e => handleCreateUser(error, loading, email, password, firstName, userRole)}
                            >
      
                              Create User
      
                            </button>

                  </>
              }

                  
               </form> 

              {Validations.setErrorMessageContainer()} 
                       
            </div> 
    );

}



export default RegistrationForm;