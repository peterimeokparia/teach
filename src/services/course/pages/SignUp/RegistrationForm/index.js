import { 
useState } from 'react';

import { 
Validations } from  'services/course/helpers/Validations';

import { 
toast } from 'react-toastify';

import './style.css';

const RegistrationForm = ({ 
  error, 
  loading,
  handleCreateUser, 
  users }) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [firstName, setFirstName] = useState(''); // change firstname to firstName on reseeding db
  const [userRole, setUserRole] = useState('');
  const [ displayOtherFormFields, setToDisplayOtherFormFields ] = useState(false);

const roles = {
  Tutor: "Tutor",
  Student: "Student" 
};

const handleSubmit = (e) => {
  e.preventDefault();
};

const getOtherFormFields = () => {
    if ( email ) {
        if ( users?.find(user => user?.email === email ) === undefined ) {
            setToDisplayOtherFormFields( true );
        } else {
            toast.error("Account already exists in our system.");
          return;
        }
    } 
};

return (    
          <div className="LoginPage"> 
              <form onSubmit={ e => handleSubmit(e)}>
             {
                <>
                  <label>  
                    Email
                    <input
                        data-cy={`email`}
                        name="email"
                        type="email"
                        value={email}
                        onChange={ e => setEmail( e.target.value ) }
                        placeholder="email"
                    >
                    </input>
                  </label>
                  {( displayOtherFormFields ) 
                      ? ""
                      : <button
                          data-cy={`submit`}
                          type="submit"
                          disabled={loading}
                          onClick={getOtherFormFields}
                      >
                        Next
                      </button>     
                  }                                                                                         
                </>
             }  
              {( displayOtherFormFields ) &&  <>
                   <label>
                    Password   
                  <input
                    data-cy={`password`}
                    name="password"
                    type="password"
                    value={password}
                    onChange={ e => setPassword( e.target.value ) }
                    placeholder="password"
                  >
                  </input>
                  </label>
                  <label>
                    First Name   
                    <input
                      data-cy={`firstName`}
                      name="firstName"
                      type="text"
                      value={firstName}
                      onChange={ e => setFirstName( e.target.value ) }
                      placeholder="first name"
                    >
                    </input>
                  </label>
                  <span className="LoginPageRadioButton">
                  <span className="leftRadioButton">
                      <label>                         
                        Student
                        <input
                          data-cy={`radio-student`}
                          name="firstName"
                          type="radio"
                          value={roles.Student}
                          onChange={ e => setUserRole( e.target.value ) }
                          placeholder="first name"
                        >
                        </input>
                      </label>                    
                  </span>                    
                 <span className="rightRadioButton">
                 <label>
                    Tutor
                    <input
                      data-cy={`radio-Tutor`}
                      name="Tutor"
                      type="radio"
                      value={roles.Tutor}
                      onChange={ e => setUserRole( e.target.value ) }
                      placeholder="first name"
                    >
                    </input> 
                  </label>  
                 </span>                                   
                  </span>
                  <div>                                
                    </div>
                      { error  && (<div className="error"> { error.message }</div>)}
                      { error  && (<div className="error"> { error }</div>)}
                        <div></div>                       
                          <button
                              data-cy={`submit`}
                              type="submit"
                              disabled={loading}
                              onClick={e => handleCreateUser({ error, loading, email, password, firstName, role: userRole })} // change firstname to firstName on reseeding db
                            >
                              Create User
                            </button>
                            { error  && (<div className="error"> { error.message }</div>)}
                            { error  && (<div className="error"> { error }</div>)}
                  </>
              }
               </form> 
              {Validations.setErrorMessageContainer()}                  
            </div> 
    );
};

export default RegistrationForm;