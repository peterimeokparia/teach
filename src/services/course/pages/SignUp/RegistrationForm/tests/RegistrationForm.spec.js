import React from 'react';
import RegistrationForm  from 'services/course/pages/SignUp/RegistrationForm';
import renderer from 'react-test-renderer';

describe('RegistrationForm', () =>  {  

  beforeEach(() => {
    jest.spyOn(Date.prototype, 'toLocaleTimeString')
        .mockImplementation(() => '12:00:00');
    jest.spyOn(Date.prototype, 'toLocaleDateString')
      .mockImplementation(() => 'AM 1/01/2021');
  });

  it('renders as expected', async () => {
    const tree = renderer.create(
        <RegistrationForm /> 
    );
      expect(tree.toJSON()).toMatchSnapshot();
  })
});




