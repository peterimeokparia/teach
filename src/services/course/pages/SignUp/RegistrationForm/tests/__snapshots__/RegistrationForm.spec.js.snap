// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm renders as expected 1`] = `
<div
  className="LoginPage"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Email
      <input
        data-cy="email"
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="email"
        value=""
      />
    </label>
    <button
      data-cy="submit"
      onClick={[Function]}
      type="submit"
    >
      Next
    </button>
  </form>
  <div>
    <div>
      <div
        className="Toastify"
      />
    </div>
  </div>
</div>
`;
